### DISCUSSION TITLE
error: template argument 2 is invalid

### DISCUSSION THREAD

**Discussion by aber68** [Arduino-IRremote/Arduino-IRremote#1263] [Help] [answered]
hello, i can't deal with this error:

c:\...\libraries\Arduino-IRremote\src/IRProtocol.hpp:126:84: error: template argument 2 is invalid
  126 | struct has_ull_print<T, void_t<decltype(std::declval<T>().print(0ull, 0))>> : std::true_type {
      |___________________________________________________________________________________^~~~~~~~~

any idea whats wrong? examples compile on ESP32-CAM

best regards
andreas

**Comment by ArminJo** [comment]
Maybe wrong compiler or wrong processor.

**Comment by aber68** [comment]
in both cases i use arduino IDE 233 with the **AI Thinker ESP32-CAM** board. the above error only appears in my user design, not in the IRremote Demo sketch. I don't have an idea whats wrong ...

this are all my includes:

```
#include <Arduino.h>
#include "FS.h"
#include "SD_MMC.h"
#include "SPI.h"
#include <WiFi.h>
#include <WebServer.h>
#include <Update.h>  	  // Web OTA
#include "Audio.h"
#include <Preferences.h>  // NV speicher
#include "IRremotePinDefinitions.h"   // just renamed the file PinDefinitionsAndMore.h
#include <IRremote.hpp>
#include "DS3231.h"       // RTC (EEPROM I2C adr 0x57)
#include <Wire.h>
#include <PCF8574.h>
#include <WiFiClientSecure.h>
#include <ArduinoJson.h>
```

**Reply by ArminJo** [reply]
Then I guess one of the other libs are not compatible.
Move the lines
```
#include "IRremotePinDefinitions.h"   // just renamed the file PinDefinitionsAndMore.h
#include <IRremote.hpp>
```
upwards until the error vanishes and report your result.

**Comment by aber68** [comment, marked as answer]
thanks for your recommendation, in the following order it compiles now without any error! <Audio.h> was the problem ...

```
#include "FS.h"
#include "SD_MMC.h"
#include "SPI.h"
#include <WiFi.h>
#include <WebServer.h>
#include <Update.h>  	  // Web OTA

#include "IRremotePinDefinitions.h"
#include <IRremote.hpp>

#include <Audio.h>    // include after IRremote!
#include <Preferences.h>  // NV speicher
#include "DS3231.h"       // RTC (EEPROM I2C adr 0x57)
#include <Wire.h>
#include <PCF8574.h>
#include <WiFiClient.h>
//#include <WiFiClientSecure.h>
#include <ArduinoJson.h>
```

**Reply by ArminJo** [reply]
Thanks for reporting 👍.
